chainer/functions/connection/linear.py

```diff
@@ -43,20 +39,25 @@ def forward(self, inputs):
         if len(inputs) == 3:
             b = inputs[2]
             y += b
+        self.retain_inputs((0, 1))  # b is not retained
         return y,
 
-    def backward(self, inputs, grad_outputs):
-        x = _as_mat(inputs[0])
-        W = inputs[1]
-        gy = grad_outputs[0]
+    def backward(self, indexes, grad_outputs):
+        x, W = self.get_retained_inputs()
+        gy, = grad_outputs
 
-        gx = gy.dot(W).astype(x.dtype, copy=False).reshape(inputs[0].shape)
-        gW = gy.T.dot(x).astype(W.dtype, copy=False)
-        if len(inputs) == 3:
-            gb = gy.sum(0)
-            return gx, gW, gb
-        else:
-            return gx, gW
+        ret = []
+        if 0 in indexes:
+            gx, = LinearFunction().apply((gy, W.T))
```

Comment:
Yes, you can use `F.linear` freely in `backward`. In the above code, I used `LinearFunction().apply()` just for slightly better efficiency (but it might be better to use the plain `F.linear`).